client/extensions/woocommerce/components/product-search/search.js

```diff
@@ -0,0 +1,165 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { filter, get, identity, map } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Token from 'components/token-field/token';
+import TokenInput from 'components/token-field/token-input';
+
+class ProductSearchField extends Component {
+	static propTypes = {
+		disabled: PropTypes.bool,
+		isBorderless: PropTypes.bool,
+		onChange: PropTypes.func,
+		onInputChange: PropTypes.func,
+		value: PropTypes.array,
+	};
+
+	getTokenValue = token => {
+		if ( 'object' === typeof token ) {
+			return token.name;
+		}
+
+		return token;
+	};
+
+	getIndexOfInput = () => {
+		return this.props.value.length;
+	};
+
+	isInputEmpty = () => {
+		return this.props.currentSearch.length === 0;
```

Comment:
because this is used for keyboard interactions, it makes sense for us to consider spaces as not-empty. `trim`ing it makes it so that backspacing when the input only contains spaces delete the previous token, not the spaces in the input.